go/vt/mysqlctl/schema.go

```diff
@@ -160,8 +159,8 @@ func ResolveTables(mysqld MysqlDaemon, dbName string, tables []string) ([]string
 }
 
 // GetColumns returns the columns of table.
-func (mysqld *Mysqld) GetColumns(dbName, table string) ([]*querypb.Field, []string, error) {
-	conn, err := getPoolReconnect(context.TODO(), mysqld.dbaPool)
+func (mysqld *Mysqld) GetColumns(ctx context.Context, dbName, table string) ([]*querypb.Field, []string, error) {
+	conn, err := getPoolReconnect(ctx, mysqld.dbaPool)
```

Comment:
Note that this only applies the Context to fetching the connection. If you want the actual query to respect Context, you can replace `conn.ExecuteFetch()` everywhere with `mysqld.executeFetchContext()`.

Reply:
hmm...who should be doing that exactly?

Reply:
Any method of `Mysqld` that wants to respect Context should use `mysqld.executeFetchContext()` as a wrapper around `conn.ExecuteFetch()` instead of calling it directly.

Reply:
Done.